dex/src/main/scala/com/wavesplatform/dex/market/CreateExchangeTransactionActor.scala

```diff
@@ -24,14 +24,15 @@ class CreateExchangeTransactionActor(createTransaction: CreateTransaction) exten
   override def receive: Receive = {
     case OrderExecutedObserved(sender, event) =>
       val sameOwner = event.counter.order.sender == event.submitted.order.sender
-      log.debug(
-        s"Execution observed at $sender for OrderExecuted(${event.submitted.order.id()}, ${event.counter.order.id()}), amount=${event.executedAmount})${if (sameOwner) " Same owner for both orders" else ""}")
+      log.debug(s"Execution observed at $sender for OrderExecuted(${event.submitted.order.id()}, ${event.counter.order
+        .id()}), amount=${event.executedAmount})${if (sameOwner) " Same owner for both orders" else ""}")
       if (sameOwner || pendingEvents.contains(event)) {
         import event.{counter, submitted}
         createTransaction(event) match {
           case Right(tx) =>
             log.info(s"Created transaction: $tx")
-            context.system.eventStream.publish(ExchangeTransactionCreated(tx))
+            val created = ExchangeTransactionCreated(tx)
+            recipients.foreach(_ ! created)
```

Comment:
Just a notice
It worked but is this ok to write tx in DB and broadcast it simultaneously? I mean, maybe it would be better to broadcast, await at Node and then write or vice versa

Reply:
A client could want to resend a transaction to the network, if it wasn't. Also it is useful for debugging. That's why we  write transactions to a local DB.
Also it is good to parallelize IO tasks.